src/Generator/Passes/DelegatesPass.cs

```diff
@@ -37,177 +20,199 @@ public DelegatesPass()
         {
-            if (!base.VisitMethodDecl(method) || !method.IsVirtual || method.Ignore)
+            if (!base.VisitParameterDecl(parameter) || parameter.Namespace == null ||
+                parameter.Namespace.Ignore)
                 return false;
-            
-            var module = method.TranslationUnit.Module;
-            AddDelegatesToDictionary(method, module);
+
+            parameter.QualifiedType = CheckType(parameter.QualifiedType, parameter);
 
             return true;
         }
-        
-        private DelegateDefinition GetExistingDelegate(Module module, string delegateString)
+
+        private QualifiedType CheckType(QualifiedType type, ITypedDecl decl)
         {
-            if (module.Libraries.Count == 0)
-                return Context.Delegates.Values.FirstOrDefault(t => t.Signature == delegateString);
+            if (type.Type is TypedefType)
+                return type;
+
+            var desugared = type.Type.Desugar();
+            if (desugared.IsDependent)
+                return type;
+
+            Type pointee = desugared.GetPointee() ?? desugared;
+            if (pointee is TypedefType)
+                return type;
 
-            DelegateDefinition @delegate = null;
-            if (new[] { module }.Union(module.Dependencies).Any(
-                m => libsDelegates.ContainsKey(m) && libsDelegates[m].TryGetValue(
-                    delegateString, out @delegate)))
+            var functionType = pointee.Desugar() as FunctionType;
+            if (functionType == null)
+                return type;
+
+            FunctionType newFunctionType = GetNewFunctionType(decl, type);
+
+            var delegateName = GetDelegateName(newFunctionType);
+            var typedef = allTypedefs.SingleOrDefault(t => t.Name == delegateName);
+            if (typedef != null)
+                return new QualifiedType(new TypedefType { Declaration = typedef });
+
+            for (int i = 0; i < functionType.Parameters.Count; i++)
+                functionType.Parameters[i].Name = $"_{i}";
+
+            TypedefDecl @delegate = GetDelegate((Declaration) decl, newFunctionType, delegateName);
+            return new QualifiedType(new TypedefType { Declaration = @delegate });
+        }
+
+        private TypedefDecl GetDelegate(Declaration decl, FunctionType newFunctionType, string delegateName)
+        {
+            DeclarationContext namespaceDelegates = GetDeclContextForDelegates(decl.Namespace);
+            var delegateType = new QualifiedType(new PointerType(new QualifiedType(newFunctionType)));
+            var access = decl is Method ? AccessSpecifier.Private : AccessSpecifier.Public;
+            var existingDelegate = allTypedefs.SingleOrDefault(t => t.Name == delegateName);
+            if (existingDelegate != null)
+            {
+                // Code to ensure that if a delegate is used for a virtual as well as something else, it finally ends up as public
```

Comment:
Remove the ` Code to ensure that` prefix on the comment, does not say anything relevant and takes up whitespace

Or just wrap the comment into more lines